src/domain/wallet-invoices/wallet-invoice-factory.types.d.ts

```diff
@@ -0,0 +1,8 @@
+interface IWalletInvoiceFactory {
```

Comment:
No need to be an `interface` right? Or are there more than 1 implementations?

Reply:
Changed :+1: 

There are a few other interfaces that only have 1 implementation as well. Should we look at changing these too?

![image](https://user-images.githubusercontent.com/17693119/129756662-1805a00a-9433-42dd-88bc-147585f93f9e.png)

Reply:
The reason I think these should be interfaces is because the implementation is in another module - so we are decoupling declaration from implementation.